synapse/storage/databases/main/room.py

```diff
@@ -1179,6 +1179,18 @@ async def upsert_room_on_join(self, room_id: str, room_version: RoomVersion):
         Called when we join a room over federation, and overwrites any room version
         currently in the table.
         """
+        # It's possible that we already have events for the room in our DB
+        # without a corresponding room entry. If we do then we don't want to
+        # mark the room as having an auth chain cover index.
+        max_ordering = await self.db_pool.simple_select_one_onecol(
+            table="events",
+            keyvalues={"room_id": room_id},
+            retcol="MAX(stream_ordering)",
+            allow_none=True,
+            desc="upsert_room_on_join",
+        )
+        has_auth_chain_index = max_ordering is None
```

Comment:
Is it worth abstracting this to a separate method `async def has_auth_chain_index(room_id: str) -> bool`?

Reply:
Done